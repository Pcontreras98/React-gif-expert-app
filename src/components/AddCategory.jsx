import { useState } from 'react';

export const AddCategory = ( {onNewCategory} ) => {

    const [inputValue, setinputValue] = useState('');
    const [selectValue, setselectValue] = useState('');
    
    const onInputChange = ( { target} ) =>{
        setinputValue(target.value);
    }

    const onSelectChange = ( { target } ) =>{
      setselectValue(target.value);
    }


    const onSubmit = (event) => {
        event.preventDefault();
        if ( inputValue.trim().length <=1 || selectValue === '' ) return;
       
        setinputValue('');
        onNewCategory(inputValue.trim(),selectValue);
        
        
    }

  return (
    <form onSubmit={ onSubmit } className='width-100 flex'>
        <input
        type = 'text'
        placeholder="Buscar gifs"
        value = { inputValue }
        onChange={ onInputChange }
        />

        <select className='option-select' value = { selectValue } onChange={ onSelectChange } >
                <option value='' disabled>Cantidad</option>
                <option value={4}>4</option>
                <option value={8}>8</option>
                <option value={12}>12</option>
                <option value={16}>16</option>
                <option value={20}>20</option>
        </select>
    </form>
  )
}