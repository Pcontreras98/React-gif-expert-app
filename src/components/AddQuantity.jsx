import { useState } from "react";

export const AddQuantity = ({ onNewQuantity }) => {
  const [selectValue, setSelectValue] = useState('');

  const handleSelectChange = (e) => {
    const value = e.target.value;
    setSelectValue(value);
    onNewQuantity(value);
  };

  return (
    <div className="title-quantity">
      <h1 className="title"> Quantity </h1>
      <select defaultValue="" className="select" onChange={handleSelectChange}>
        <option value="" disabled>Seleccione</option>
        <option value="8">8</option>
        <option value="12">12</option>
        <option value="16">16</option>
        <option value="20">20</option>
      </select>
    </div>
  );
};
