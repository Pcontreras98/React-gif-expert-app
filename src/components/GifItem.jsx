import PropTypes from "prop-types";

export const GifItem = ({ title , url }) => {

  

  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p className="poppins-regular"> {title} </p>
    </div>
  )
}

GifItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

