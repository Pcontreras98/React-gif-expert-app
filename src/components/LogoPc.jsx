import React from 'react'
import imagen from '../image/pcLogo.png'

export const LogoPc = () => {
  return (
    <div className='pc-logo-contenedor'>
         <img 
                className='pc-logo'
                src={imagen}
            />   
    </div>
  )
}
