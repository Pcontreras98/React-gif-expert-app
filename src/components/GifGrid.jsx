import { GifItem } from './GifItem';
import { useFetechGifs } from '../hooks/useFetchGifs';



export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetechGifs( category );

    return (
        <>  
            <h3 className='title-category'> {category} </h3>
            {
                isLoading && ( <h2>Cargando...</h2> )
            }
            
            <div className='card-grid'>
            {             
                images.map(( image ) => (
                        <GifItem key={ image.id }
                                { ...image}
                        />
                    ))
                }
            </div>

        </>
    )
}
