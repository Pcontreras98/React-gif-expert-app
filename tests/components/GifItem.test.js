import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas en GifItem', () => { 

    const title = 'Goku';
    const url = 'https://goku/goku2.com';


    test('Debe hacer match con el snapshot', () => { 

      const { container } = render( <GifItem title = { title } url = { url } />);
      expect ( container ).toMatchSnapshot();
      
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => { 
      
      render( <GifItem title = { title } url = { url } />);
      //screen.debug(); 
      const { src, alt } = screen.getByRole('img');
      expect( src ).toBe(url);
      expect( alt ).toBe(alt);

    }); 

    test('Debe mostrar el titulo en el componente', () => { 
      
      render( <GifItem title = { title } url = { url } />);
      expect( screen.getByText( title )).toBeTruthy();  

    }); 
      
    
});