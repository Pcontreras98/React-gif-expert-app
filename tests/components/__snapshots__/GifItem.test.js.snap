// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku/goku2.com"
    />
    <p
      class="poppins-regular"
    >
       
      Goku
       
    </p>
  </div>
</div>
`;
